dotnet/src/Extensions/PromptTemplates.Liquid/LiquidPromptTemplate.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Scriban;
+
+namespace Microsoft.SemanticKernel.PromptTemplates.Liquid;
+internal class LiquidPromptTemplate : IPromptTemplate
+{
+    private readonly PromptTemplateConfig _config;
+
+    public LiquidPromptTemplate(PromptTemplateConfig config)
+    {
+        if (config.TemplateFormat != LiquidPromptTemplateFactory.LiquidTemplateFormat)
+        {
+            throw new ArgumentException($"Invalid template format: {config.TemplateFormat}");
+        }
+
+        this._config = config;
+    }
+
+    public Task<string> RenderAsync(Kernel kernel, KernelArguments? arguments = null, CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(kernel);
+        var template = this._config.Template;
+        var liquidTemplate = Template.ParseLiquid(template);
+        var nonEmptyArguments = arguments.Where(x => x.Value is not null).ToDictionary(x => x.Key, x => x.Value!);
+        var renderedResult = liquidTemplate.Render(nonEmptyArguments);
+
+        // parse chat history
+        // for every text like below
+        // (system|assistant|user|function):
+        // xxxx
+        //
+        // turn it into
+        // <Message role="system|assistant|user|function">
+        // xxxx
+        // </Message>
+
+        var roleRegex = new Regex(@"(?<role>system|assistant|user|function):[\s]+");
```

Comment:
This should be cached into a static readonly field rather than constructed on each use.